portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -24,9 +27,49 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  private List<String> commentParts;
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    String json = convertToJsonUsingGson(commentParts);
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the form.
+    String name = getParameter(request, "name", "Anonymous");
+    String email = getParameter(request, "email", "n/a");
+    String comment = getParameter(request, "comment-input", "");
+
+    // Add input into an ArrayList.
+    commentParts = new ArrayList<>();
```

Comment:
so essentially there can only be at most one comment right, since everytime someone does a 'post' the `commentParts` gets re-initialized with the new empty array list

Reply:
Yup, I think the point of this PR/this part of the tutorial is just to get an HTML form running.